spring-kotlin/aws-ssh-server-manager/src/main/kotlin/dev/santos/awssshservermanager/dto/CreateTenantDto.kt

```diff
@@ -0,0 +1,21 @@
+package dev.santos.awssshservermanager.dto
+
+import dev.santos.awssshservermanager.model.Tenant
+import javax.validation.constraints.Pattern
+import javax.validation.constraints.Size
+
+data class CreateTenantDto(
+  @field:Pattern(regexp = "^[a-z]+(-[a-z\\d]+)*\$")
+  @field:Size(min=2, max=50)
+  val name: String="",
+  val awsApiKey: String="",
+  val awsApiSecret: String=""
+) {
+  fun toTenant(): Tenant {
+    return Tenant(
+      name = this.name,
+      awsApiKeySsmName = "${this.name}-api-key",
+      awsApiSecretSsmName = "${this.name}-api-secret"
+    )
```

Comment:
Is it a good practice to have logic within the DTO? I think that's the job of a service that will get the `CreateTenantDto` and create the Tenant

Reply:
Short answer: it is not. Thinking about it, Mapper would be more suitable!